pkg/controller/storage/mem/storage_instance.go

```diff
@@ -37,37 +37,58 @@ func newMemStorageInstance() *memStorageInstance {
 }
 
 func (m *memStorageInstance) List() ([]*servicecatalog.Instance, error) {
-	ret := make([]*servicecatalog.Instance, len(m.instances))
+	copy := make([]*servicecatalog.Instance, len(m.instances))
 	i := 0
-	for _, val := range m.instances {
-		ret[i] = val
+	for _, inst := range m.instances {
+		copy[i] = &servicecatalog.Instance{}
+		if err := deepCopy(inst, copy[i]); err != nil {
+			return nil, err
+		}
 		i++
 	}
-	return ret, nil
+	return copy, nil
 }
 
 func (m *memStorageInstance) Get(name string) (*servicecatalog.Instance, error) {
-	ret, ok := m.instances[name]
+	inst, ok := m.instances[name]
 	if !ok {
 		return nil, errNoSuchInstance
 	}
-	return ret, nil
+	copy := &servicecatalog.Instance{}
+	if err := deepCopy(inst, &copy); err != nil {
+		return nil, err
+	}
+	return copy, nil
 }
 
 func (m *memStorageInstance) Create(inst *servicecatalog.Instance) (*servicecatalog.Instance, error) {
 	if _, err := m.Get(inst.Name); err == nil {
 		return nil, errInstanceAlreadyExists
 	}
-	m.instances[inst.Name] = inst
-	return inst, nil
+	m.instances[inst.Name] = &servicecatalog.Instance{}
+	if err := deepCopy(inst, m.instances[inst.Name]); err != nil {
+		return nil, err
```

Comment:
same comment here as in https://github.com/kubernetes-incubator/service-catalog/pull/261/files#r96036637